src/back/space/src/service/service.cpp

```diff
@@ -323,4 +329,43 @@ bool Service::isKeyReserved(const std::string& key) {
 	return reserved.contains(key);
 }
 
+tokens::Tokens& Service::Tokens() {
+	return _tokens;
+}
+
+model::Space Service::GetByKeyIfAdmin(const std::string& key, const std::string userId) {
+	model::Space space = _repo.Space().SelectByKey(key);
+	if (_repo.SpaceUser().IsAdmin(space.id, userId))
+		return space;
+	else
+		throw errors::Forbidden403();
+}
+
+std::string Service::GetKeyFromHeader(const std::string& header) {
+	std::regex rgx("^/([^/]+)/");
+	std::smatch match;
+
+	if (!std::regex_search(header.begin(), header.end(), match, rgx))
+		throw errors::BadRequest400("Space key is missing");
+
+	return match[1];
+}
+
+std::string Service::GenerateCookieName(const std::string& key) {
+	uint32_t crc32 = generateCRC32(key);
+	std::string cookieName = "space_" + std::to_string(crc32);
+	return cookieName;
+}
+
+std::string Service::CreateToken(const std::string& id, const std::string& key, const std::string& userId, const std::string& role) {
+	std::string token = Tokens().Create(key, id, role, userId, _tokenExpireSecs);
+	return token;
+}
+
+uint32_t Service::generateCRC32(const std::string& data) {
+	boost::crc_32_type result;
+    result.process_bytes(data.data(), data.length());
+    return result.checksum();
```

Comment:
Замени пожалуйста пробелы на табы
```suggestion
	result.process_bytes(data.data(), data.length());
	return result.checksum();
```

Reply:
Готово.